docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -39,12 +39,34 @@ impl Storage {
 If you have defined a `Storage` struct following this naming scheme, then it will be made available to you through the reserved `storage` keyword within your contract functions.
 
 :::warning Using slot `0` is not supported!
-No storage values should be initialized at slot `0`. This is a known issue that will be fixed in the future.
+No storage values should be initialized at slot `0` - storage slots begin at `1`. This is a known issue that will be fixed in the future.
 :::
 
+## Storage Slots
+
+Each contract's state is represented with a storage slot key value store of type `Map<Field, Field>`. For now, storage slot positions for each variable must be explicitly assigned inside the `Storage impl`. Except for `Map` types, storage is in contiguous blocks, so you must calculate how many slots each public variable requires, and increment the next variable's slot by this amount.
+
+When assigning contract storage slots, `Map`s are treated as occupying only 1 storage slot (its "base_slot"), because the actual values are stored in derived slots calculated as `map_value_storage_slot = pedersen_hash(base_slot, key)`.
+
+:::info
+Private variables only require one single slot, because all notes are linked to contract state through a single `storage slot` attribute in their header.
+:::
+
+We currently do not support any "packing" type optimizations as in most EVM languages.
+
+Storage in Aztec is implemented as a single global merkle tree, with each contract's internal storage slot combined with its contract address to generate its position in the global tree as `global_storage_slot = pedersen_hash(contract_storage_slot, contract_address)`.
```

Comment:
the storage slot is used to make requests from the RPC server (ie, get me notes for this variable), but it can return multiple notes.  even though they have different "storage slots" in the UTXO tree

Reply:
and we grab those private notes by the header (contract address + storage slot + nonce)